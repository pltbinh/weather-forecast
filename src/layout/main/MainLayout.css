.content {
  min-height: 420px;
}

.header {
  color: white;
  background-image: url('../../asset/header.jpg');
  text-align: center;
}

.footer {
  position:absolute;
  bottom:0;
  width: 100%;
  background: #262936;
  color: white;
  font-size: 14px;
  min-height: 48px;
}
