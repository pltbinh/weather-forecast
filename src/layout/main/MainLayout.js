import React from 'react'
import { Jumbotron } from 'react-bootstrap'

import './MainLayout.css'
export default function MainLayout({ children }) {
  return (
    <>
      <header data-testid='mainHeader'>
        <Jumbotron fluid className='header'>
          <h1>Weather Forecast Application</h1>
        </Jumbotron>
      </header>
      <div className='content'>
        {children}
      </div>
      <footer data-testid='mainFooter'>
        <div className='footer mt-3'>
          <p>Designed by pltbinh</p>
        </div>
      </footer>
    </>
  )
}