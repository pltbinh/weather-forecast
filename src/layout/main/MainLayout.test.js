import React from 'react';
import { screen, render } from '@testing-library/react'
import renderer from 'react-test-renderer'

import MainLayout from './MainLayout';

describe('FilterComponent', () => {
  it('should render correctly', () => {
    // act
    const component = renderer.create(<MainLayout />)
    const tree = component.toJSON()

    // assert
    expect(tree).toMatchSnapshot()
  });

  it('should render correct footer and header', () => {
    // act
    render(<MainLayout />)

    const header = screen.getByTestId('mainHeader')
    const footer = screen.getByTestId('mainFooter')

    // assert
    expect(header).toBeDefined()
    expect(footer).toBeDefined()
  });
})