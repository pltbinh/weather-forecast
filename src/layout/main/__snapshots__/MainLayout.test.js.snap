// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterComponent should render correctly 1`] = `
Array [
  <header
    data-testid="mainHeader"
  >
    <div
      className="header jumbotron jumbotron-fluid"
    >
      <h1>
        Weather Forecast Application
      </h1>
    </div>
  </header>,
  <div
    className="content"
  />,
  <footer
    data-testid="mainFooter"
  >
    <div
      className="footer mt-3"
    >
      <p>
        Designed by pltbinh
      </p>
    </div>
  </footer>,
]
`;
